src/days-2-4-adv/adv.py

```diff
@@ -49,3 +50,41 @@
 # Print an error message if the movement isn't allowed.
 #
 # If the user enters "q", quit the game.
+while True:
+    print(f'\nLocation: {player.location.name}')
+    print(f'{player.location.description}')
+    if len(player.location.items) > 0:
+        player.location.print_items()
+
+    player_input = input(f'\nWhat does {player.name} do?   ')
+    player_input_args = player_input.split(' ')
+
+    if len(player_input_args) == 0:
+        print('\nControls: N,S,E,W to move to a different room. T to take an item. D to drop an item. I to view your inventory. Q to quit')
+
+    if len(player_input_args) == 1:
+        if player_input[0].lower() == "q":
+            break
+        elif player_input[0].lower() == "n" or player_input[0].lower() == "s" or player_input[0].lower() == "w" or player_input[0].lower() == "e":
+            player.change_location(player_input[0])
+        elif player_input[0].lower() == "i":
+                player.view_inventory()
+        else:
+            print('\nControls: N,S,E,W to move to a different room. T to take an item. D to drop an item. I to view your inventory. Q to quit')
+
+    if len(player_input_args) == 2:
+        if player_input[0].lower() == "t" or player_input[0].lower() == "take":
+            taking_item = player.location.find_item(player_input_args[1])
+            if taking_item == None:
+                print(f"You couldn't find a {player_input[1]}")
+            else:
+                player.take_item(taking_item)
+                player.location.remove_item(taking_item)
+
+        elif player_input[0].lower() == "d" or player_input[0].lower() == "drop":
+            dropped_item = player.find_item(player_input_args[1])
+            if dropped_item == None:
+                print(f"You dont have a {player_input_args[1]} to drop!")
+            else:
+                player.drop_item(dropped_item)
+                player.location.add_item(dropped_item)
```

Comment:
This looks logically correct at first glance but when I run the game, sometimes it will list an item in my inventory, but when I try to drop it, it says I don't have an item to drop. This happens intermittently - sometimes it works and sometimes it doesn't. Not sure why yet.

Reply:
Maybe the bug is in line 85? Does the player.find_item refer to the inventory or something that the player just picked up??